eng/jobs/run-publish-project.yml

```diff
@@ -0,0 +1,120 @@
+parameters:
+  projectName: ''
+  dependency: null
+
+jobs:
+
+- template: /eng/common/templates/post-build/setup-maestro-vars.yml
+
+- job: CustomPublish
+  displayName: Custom Publish
+  dependsOn: setupMaestroVars
+  # Only run this job if setup-maestro-vars says the current branch publishes to this channel.
+  # Logic copied from ../common/templates/post-build/channels/netcore-dev-5.yml
+  condition: contains(
+    dependencies.setupMaestroVars.outputs['setReleaseVars.InitialChannels'],
+    format('[{0}]', variables['${{ parameters.dependency.channel.bar }}']))
```

Comment:
I'm surprised that this works!

Reply:
What about it? I copy pasted the code that Arcade uses to determine whether to run a job depending on what default channels the branch has.